### PR TITLE
[REST Server] Show err.stack when job config is not valid.

### PR DESCRIPTION
**Pull request by hwuu**

### CODE REVIEW

File: rest-server/src/middlewares/parameter.js [hunk 1]
```diff
@@ -20,18 +20,19 @@
...
 /**
  * Validate parameters.
  */
 const validate = (schema) => {
   return (req, res, next) => {
     Joi.validate(req.body, schema, (err, value) => {
       if (err) {
-        logger.warn('parameters validation error\n%s', err.stack);
+        errorType = 'ParameterValidationError';
+        errorMessage = 'Could not validate request data.\n' + err.stack;
```
Comment (abuccts) [inline rest-server/src/middlewares/parameter.js]: variable not defined

**Comment by abuccts** [comment]
Do we really need to show error stack to user?
If so, set `NODE_ENV` to `development` is enough.

**Comment by abuccts** [comment]
Please refer to [rest-server/src/config/express.js#L52-L59](https://github.com/Microsoft/pai/blob/master/rest-server/src/config/express.js#L52-L59) for error handler.

File: rest-server/src/middlewares/parameter.js [hunk 2]
```diff
@@ -20,18 +20,19 @@
...
+        logger.warn('[%s] %s', errorType, errorMessage);
         return res.status(500).json({
-          error: 'ParameterValidationError',
-          message: 'could not validate request data'
+          error: errorType,
+          message: errorMessage
```
Comment (abuccts) [inline rest-server/src/middlewares/parameter.js]: please add trailing comma to pass lint check

Comment (hwuu) [reply rest-server/src/middlewares/parameter.js]: Will do.

**Comment by hwuu** [comment]
@abuccts : I think we should provide users, rather than us, sufficient hint to help them spot the causes of errors. If we rely on the 'env' config, we should restart the service to make it take effect.

**Pull request merged**